### DISCUSSION TITLE
ERC-721, Doubt !!

### DISCUSSION THREAD

**Discussion by yasharyan0904** [Cyfrin/foundry-full-course-cu#1918] [Q&A] [answered]
event ApprovalForAll(address indexed _owner, address indexed _operator, bool _approved);
Refer-> https://eips.ethereum.org/EIPS/eip-721

What is the use of this operator, like why we need a operator to do stuff on the behalf of the owner of the NFTs?
Why the owner can't approve and transfer NFTs on its own?

**Comment by EngrPips** [comment, marked as answer]
Hello @yasharyan0904, Let's play a scenario where you listed your NFT on a marketplace such as `opensea,` but you didn't approve them to send your token, So after a few days, `opensea` got a buyer for your NFT who is ready to pay instantly. What do you think opensea should do to help you transfer the NFT to the buyer? Call your mobile number? Send you an email saying, "Hey, we got a buyer. Please transfer the NFT to them now."? obviously, none of those make sense. Hence, you need to approve them to be able to transfer your NFT so that they can easily transfer your NFT to the buyer and send the payment to your address.

**Reply by yasharyan0904** [reply]
Good scenario to explain :) , Thanks I got it.

**Reply by EngrPips** [reply]
Interesting, I am glad you understand it. Keep crushing it!
